src/libs/LocalePhoneNumber.ts

```diff
@@ -18,6 +19,10 @@ function formatPhoneNumber(number: string): string {
         return '';
     }
 
+    // do not parse the string, if it doesn't contain the SMS domain and it's not a phone number
+    if (number.indexOf(CONST.SMS.DOMAIN) === -1 && !CONST.REGEX.DIGITS_AND_PLUS.test(number)) {
```

Comment:
I wonder if we could improve this at compile time by creating separate types for `EmailLogin` and `SMSLogin`, and making this function accept `SMSLogin` only. Then remove the check in this low-level function? thoughts?

Reply:
I am not that much familiar with business logic, so I might fancy some insights. From what I understand, we only have one property that holds either Email or Sms/number. How can we differentiate between them? 

I don't know whether we get this property from backend Or we compile the data on front end. In case of latter, we can check whether the property is an email or sms using the logic we have in this PR and add a new property like `isEmail`. Then in `formatPhoneNumber` we can just check if it's email using the new property, we will early return.

But the issue is we will need to pass the `isEmail` property to the function wherever it's called. And just looking in VScode search, it's ~40 places so a lot of refactor. I guess we might be safe with what we have in this PR because it doesn't introduce such huge refactor.

Did I understand correctly?

Reply:
> From what I understand, we only have one property that holds either Email or Sms/number. How can we differentiate between them?

We could check _once_ at a higher level if a login is an SMS login and narrow the type accordingly.

Less passionate about changing anything here - I think the most important change I'd like to see done in this PR is to move the memoization into the existing localization lib rather than adding "CommonTranslations"